### PR TITLE
Created plot_sampling_hist function and tests

### PR DESCRIPTION
**Pull request by YueJiangMDSV**
Hi Team,

In this pr, 
1. created plot_sampling_hist
2. created tests for plot_sampling_hist

Please review it. Thanks!

Alex

### CODE REVIEW

File: samplingsimulatorpy/test_plot_sampling_hist.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+def test_plot_sampling_hist():
+    """
+    This function tests the plot_sampling_hist function.
+    
+    Returns:
+    --------
+    None
+        The test should pass and no asserts should be displayed. 
+    """
```
Comment (tguo9) [inline samplingsimulatorpy/test_plot_sampling_hist.py]:
```suggestion
```
Probably no needs of docstring here

File: samplingsimulatorpy/plot_sampling_hist.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    
+    summary = samples.groupby(['replicate', 'size', 'rep_size']).mean()
+    summary = summary.reset_index()
+    x_max = summary.iloc[:,3].max()
+    x_min = summary.iloc[:,3].min()
+    p = alt.Chart(summary).mark_bar().encode(
+        x=alt.X(f"{summary.columns[3]}:Q", title = "mean of sampling distribution",bin = alt.Bin(extent=[x_min, x_max], step=1/10)),
+        y=alt.Y('count()')).facet(
+        facet='size', columns = 2)
+    return p
```
Comment (tguo9) [inline samplingsimulatorpy/plot_sampling_hist.py]:
```suggestion
    return p
    
```
A new line at the end of file

File: samplingsimulatorpy/test_plot_sampling_hist.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    # Check x-axis is mapped to the correct variable and plotting correctly
+    assert test_dict['spec']['mark'] == 'bar', "the plot type is correct"
+    assert test_dict['spec']['encoding']['x']['type'] == 'quantitative', "x-axis should have quantitivate values" 
+    assert test_dict['spec']['encoding']['x']['field'] == 'var_name', "x-axis is mapped to the wrong variable"
+    
+    
+    # Check y-axis is mapped to the correct variable and plotting correctly
+    assert test_dict['spec']['encoding']['y']['type'] == 'quantitative', "y-axis should have quantitivate values"
+    assert test_dict['spec']['encoding']['y']['aggregate'] == 'count', "y-axis should represent aggregate count data"
+
```
Comment (tguo9) [inline samplingsimulatorpy/test_plot_sampling_hist.py]:
```suggestion
```
No need for test file have a new line at the end of file

File: samplingsimulatorpy/plot_sampling_hist.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    """
+    
+    # check the validity of input
+    if not isinstance(samples, pd.DataFrame):
+        raise TypeError("'samples' should be a dataframe")
+    
+    if not np.issubdtype(samples.to_numpy().dtype, np.number):
+        raise ValueError("'samples' should only contain numeric values")
+        
+    if not x.shape[1]==4:
```
Comment (tguo9) [inline samplingsimulatorpy/plot_sampling_hist.py]:
```suggestion
    if not samples.shape[1]==4:
```
x is not defined, I'm guessing it should be samples?

Comment (YueJiangMDSV) [reply samplingsimulatorpy/plot_sampling_hist.py]: Good catch! I'll update that

File: samplingsimulatorpy/test_plot_sampling_hist.py (hunk 1 above)
Comment (YueJiangMDSV) [reply samplingsimulatorpy/test_plot_sampling_hist.py]: Good to know! Thanks!

File: samplingsimulatorpy/plot_sampling_hist.py (hunk 2 above)
Comment (YueJiangMDSV) [reply samplingsimulatorpy/plot_sampling_hist.py]: Just deleted the new line

File: samplingsimulatorpy/test_plot_sampling_hist.py (hunk 3 above)
Comment (YueJiangMDSV) [reply samplingsimulatorpy/test_plot_sampling_hist.py]: Deleted the new line

**Review by tguo9 — APPROVED** [review]
Looks good! Approve

**Pull request merged**